### DISCUSSION TITLE
issue with proposed automation for triggering ring alarm base

### DISCUSSION THREAD

**Discussion by jonny9998** [tsightler/ring-mqtt#490] [General Discussion]
First of all, this is a very well done integration.  Thankyou.

I am new to this, so please forgive me for any obvious errors I am not seeing.

I made this code in HA to disarm the alarm base if a motion detector near my bedroom is triggered.  Ie.  If I get out of bed to let the dog out and forget in my foggy brain to disarm the panel before opening the door:

id: '1667630853312'
alias: disarm alarm if zwave motion detected
description: ''
trigger:
  - type: motion
    platform: device
    device_id: e05f59ac983107abb64c5cd3b8d28f6b
    entity_id: binary_sensor.main_hallway_motion_sensor_motion_detection
    domain: binary_sensor
condition: []
action:
  - service: alarm_control_panel.alarm_disarm
    data: {}
    target:
      entity_id: alarm_control_panel.prince_george_alarm
mode: single

This code works no problem.

I am now trying to make a similar automation that will trigger the alarm if that same sensor is triggered when I am not home.

The code looks like this:

alias: Trigger alarm if not home
description: ""
trigger:
  - type: motion
    platform: device
    device_id: e05f59ac983107abb64c5cd3b8d28f6b
    entity_id: binary_sensor.main_hallway_motion_sensor_motion_detection
    domain: binary_sensor
condition:
  - condition: state
    entity_id: alarm_control_panel.prince_george_alarm
    state: armed_home
action:
  - service: alarm_control_panel.alarm_trigger
    data: {}
    target:
      entity_id: alarm_control_panel.prince_george_alarm
mode: single

Even though this automation fires, the base does not start ringing.  Where am I going wrong?

**Comment by sktaylortrash** [comment]
I have to ask why are you massively over-complicating things by doing this as an HA automation? This is exactly the scenario Home and Away modes are for.  You exclude the sensor from Home mode and turn that on at night and include it in Away mode and set that when you leave.

**Reply by jonny9998** [reply]
Fair question.  I have just installed Ring into my home, but I already have multiple motion sensors running through through z-wave, not through Ring.  I am hoping not to have to replace these sensors with ring sensors.  Rather, I would like to integrate the sensors into the Ring system using the MQTT broker.

The real question here is how can I trigger the alarm base using HA?  My code to disarm the base works fine.  But the trigger call doesn't.  What am I missing?

**Reply by tsightler** [reply]
> The real question here is how can I trigger the alarm base using HA? My code to disarm the base works fine. But the trigger call doesn't. What am I missing?

You are missing the fact that there is no way to trigger the Ring alarm via the Ring API, other than via the panic switches.  Ring-mqtt can only perform the same functions that are possible in the Ring app, so, for example, you can test the siren on the base station, but testing the siren is not the same as triggering the alarm, and you can't test the siren while the system is armed.

If you enable the panic switches configuration option, then you can trigger burglar or fire alarms via automation of those switches, but I've been told those switches are only available if you pay for monitoring, and, realistically, if you to trigger these via automation from 3rd party sensors, it's probably a violation of the Ring monitoring terms of service.

To me it feels like you want to build a homebrew alarm with your own sensors, I'd suggest that Ring alarm isn't the best platform for this.

**Reply by jonny9998** [reply]
I was starting to come to the same conclusion.  Thanks for straightening me out on this.

**Comment by sktaylortrash** [comment]
Non-Ring sensors make sense.  
That being said and the project's author can correct me if I'm wrong. But I think to trigger the alarm from an unsupported device you need to use the Panic Button and that requires a Ring Protect plan.
